pandas/io/gbq.py

```diff
@@ -148,6 +151,43 @@ def get_credentials(self):
 
         return credentials
 
+    def get_credentials_for_service_account(self):
```

Comment:
I would do this in ``.get_credentials`` (where you have the if), no reason to have a separate function for this (when the other method is IN that function)